cnf-certification-test/certification/suite.go

```diff
@@ -151,63 +148,67 @@ func testAllOperatorCertified(env *provider.TestEnvironment, validator certdb.Ce
+				AddField(testhelper.OCPVersionType, ocpMinorVersion).
+				AddField(testhelper.OCPChannelType, channel))
 		}
 	}
-	if testFailed {
-		ginkgo.Fail("At least one operator was not certified to run on this version of OpenShift. Check Claim.json file for details.")
-	}
+	testhelper.AddTestResultReason(compliantObjects, nonCompliantObjects, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 
 func testHelmCertified(env *provider.TestEnvironment, validator certdb.CertificationStatusValidator) {
 	helmchartsReleases := env.HelmChartReleases
 	testhelper.SkipIfEmptyAny(ginkgo.Skip, helmchartsReleases)
 	// Collect all of the failed helm charts
-	failedHelmCharts := [][]string{}
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
 	for _, helm := range helmchartsReleases {
 		if !validator.IsHelmChartCertified(helm, env.K8sVersion) {
-			failedHelmCharts = append(failedHelmCharts, []string{helm.Chart.Metadata.Version, helm.Name})
-			tnf.ClaimFilePrintf(
-				"Helm Chart %s version %s is not certified.",
-				helm.Name,
-				helm.Chart.Metadata.Version,
-			)
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewHelmChartReportObject(helm.Namespace, helm.Name, "helm chart is not certified", false).
+				SetType(testhelper.HelmVersionType).
+				AddField(testhelper.Version, helm.Chart.Metadata.Version))
+			tnf.ClaimFilePrintf("Helm Chart %s version %s is not certified.", helm.Name, helm.Chart.Metadata.Version)
 		} else {
-			logrus.Infof(
-				"Helm Chart %s version %s is certified.",
-				helm.Name,
-				helm.Chart.Metadata.Version,
-			)
+			logrus.Infof("Helm Chart %s version %s is certified.", helm.Name, helm.Chart.Metadata.Version)
+			compliantObjects = append(compliantObjects, testhelper.NewHelmChartReportObject(helm.Namespace, helm.Name, "helm chart is certified", true).
+				SetType(testhelper.HelmVersionType).
+				AddField(testhelper.Version, helm.Chart.Metadata.Version))
 		}
 	}
-	testhelper.AddTestResultLog("Non-compliant", failedHelmCharts, tnf.ClaimFilePrintf, ginkgo.Fail)
+	testhelper.AddTestResultReason(compliantObjects, nonCompliantObjects, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 
 func testContainerCertificationStatusByDigest(env *provider.TestEnvironment, validator certdb.CertificationStatusValidator) {
-	failedContainers := []configuration.ContainerImageIdentifier{}
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
 	for _, c := range env.Containers {
 		if c.ContainerImageIdentifier.Name == "" || c.ContainerImageIdentifier.Repository == "" {
 			tnf.ClaimFilePrintf("Container name = %q or repository = %q is missing, skipping this container to query", c.ContainerImageIdentifier.Name, c.ContainerImageIdentifier.Repository)
 			continue
 		}
+		//nolint:gocritic
```

Comment:
To avoid de nolint, you should probably use a switch here, which is the idiomatic way in golang:
```
switch {
case c.ContainerImageIdentifier.Digest == "":
  ...
case !testContainerCertification(c.ContainerImageIdentifier, validator):
  ...
default:
  ...
}
```

Reply:
Done.